### DISCUSSION TITLE
Logging additional content

### DISCUSSION THREAD

**Discussion by slowMelon01** [ottowayi/pycomm3#161] [Q&A] [answered]
I'm trying to log additional messages to my log file other than the default ones set out in the pycomm3 module.
However, I can't figure out how to do this. 

Here's a code example:
```
import logging
from pycomm3.logger import configure_default_logger, LOG_VERBOSE

configure_default_logger(level=LOG_VERBOSE, filename='./logging.log')

def discoverPLCs(): # Function to discover any PLC on the network
    ips, slots, progName = [], [], []
    try:
        discovery = CIPDriver.discover() # Return list of all CIP devices on the network
        for device in discovery: # Go through discovery list and append any PLC#'s to a list
            if device['product_type'] == "Programmable Logic Controller":
                ips.append(device['ip_address'])        
        if len(ips) > 0: # Print the discovered PLC's, if there are any
            ...
        else:
            logging.info("No PLC's discovered on the network") # THIS DOES NOT APPEAR IN MY LOG FILE
    except Exception:
        traceback.print_exc()
```
This is the contents of the log file after running function `discoverPLCs`:
> 2021-07-06 22:19:36,581 [INFO] pycomm3.cip_driver.CIPDriver.discover(): Discovering devices...
> 2021-07-06 22:19:36,595 [DEBUG] pycomm3.cip_driver.CIPDriver.discover(): Broadcasting discover for IP: 10.245.188.1
> 2021-07-06 22:19:37,599 [DEBUG] pycomm3.cip_driver.CIPDriver.discover(): Broadcasting discover for IP: 136.129.0.69
> 2021-07-06 22:19:38,613 [DEBUG] pycomm3.cip_driver.CIPDriver.discover(): Broadcasting discover for IP: 169.254.132.173
> 2021-07-06 22:19:39,617 [DEBUG] pycomm3.cip_driver.CIPDriver.discover(): Broadcasting discover for IP: 169.254.112.181
> 2021-07-06 22:19:40,627 [DEBUG] pycomm3.cip_driver.CIPDriver.discover(): Broadcasting discover for IP: 192.168.232.1
> 2021-07-06 22:19:41,636 [DEBUG] pycomm3.cip_driver.CIPDriver.discover(): Broadcasting discover for IP: 192.168.230.1
> 2021-07-06 22:19:42,646 [DEBUG] pycomm3.cip_driver.CIPDriver.discover(): Broadcasting discover for IP: 192.168.0.34
> 2021-07-06 22:19:43,651 [DEBUG] pycomm3.cip_driver.CIPDriver.discover(): Broadcasting discover for IP: 169.254.166.141
> 2021-07-06 22:19:44,658 [DEBUG] pycomm3.cip_driver.CIPDriver.discover(): No devices found so far, attempting broadcast without binding to an IP.
> 2021-07-06 22:19:45,671 [INFO] pycomm3.cip_driver.CIPDriver.discover(): No Ethernet/IP devices discovered

Where am I going wrong?

**Comment by ottowayi** [comment, marked as answer]
You're not, that function is only configuring the `pycomm3` logger where as you're using the root logger.  I didn't even think about custom loggers when I added that function, so I just did.  If you update to the latest release (1.2.1), you can pass in the name of the logger you also want to configure along with the pycomm3 one.  In your case where you're using the root logger, just pass in an empty string (e.g. `configure_default_logger(level=LOG_VERBOSE, filename='./logging.log', logger='')`.  Hope that works for you.
